codeworld-server/src/Main.hs

```diff
@@ -354,6 +358,31 @@ runHandler = do
     modifyResponse $ setContentType "text/javascript"
     serveFile (buildRootDir mode </> targetFile programId)
 
+exportAndroidHandler :: Snap()
+exportAndroidHandler = do
+    mode <- getBuildMode
+    Just source <- getParam "source"
+    maybeAppName <- getParam "appName"
+    let appName = BC.unpack $ fromMaybe (BC.pack "CodeWorld App") maybeAppName 
```

Comment:
Is there ever a case you expect to get a request without an app name?  If not, just fail the pattern match in that case.